docs/assets/other/json/apisix-grafana-dashboard.json

```diff
@@ -1622,6 +1622,216 @@
+      "targets": [
+        {
+          "expr": "sum(apisix_shared_dict_capacity_bytes{name=~\"internal-status|upstream-healthcheck|worker-events|lrucache-lock|balancer-ewma|balancer-ewma-locks|balancer-ewma-last-touched-at|etcd-cluster-health-check\", instance=~\"$instance\"}) by (name)",
+          "instant": false,
+          "interval": "",
+          "intervalFactor": 1,
+          "legendFormat": "{{state}}",
+          "refId": "A"
+        }
+      ],
+      "thresholds": [],
+      "timeFrom": null,
+      "timeRegions": [],
+      "timeShift": null,
+      "title": "Nginx shared dict capacity",
+      "tooltip": {
+        "shared": true,
+        "sort": 0,
+        "value_type": "individual"
+      },
+      "type": "graph",
+      "xaxis": {
+        "buckets": null,
+        "mode": "time",
+        "name": null,
+        "show": true,
+        "values": []
+      },
+      "yaxes": [
+        {
+          "$$hashKey": "object:420",
+          "decimals": null,
+          "format": "decbytes",
+          "label": "",
+          "logBase": 1,
+          "max": null,
+          "min": null,
+          "show": true
+        },
+        {
+          "$$hashKey": "object:421",
+          "decimals": null,
+          "format": "Misc",
+          "label": "",
+          "logBase": 1,
+          "max": null,
+          "min": null,
+          "show": true
+        }
+      ],
+      "yaxis": {
+        "align": false,
+        "alignLevel": null
+      }
+    },
+    {
+      "aliasColors": {},
+      "bars": false,
+      "dashLength": 10,
+      "dashes": false,
+      "datasource": "${DS_PROMETHEUS}",
+      "description": "The free space of each nginx shared DICT since APISIX start",
+      "fieldConfig": {
+        "defaults": {
+          "custom": {},
+          "links": []
+        },
+        "overrides": []
+      },
+      "fill": 1,
+      "fillGradient": 0,
+      "gridPos": {
+        "h": 7,
+        "w": 12,
+        "x": 12,
+        "y": 57
+      },
+      "hiddenSeries": false,
+      "id": 35,
+      "legend": {
+        "alignAsTable": false,
+        "avg": false,
+        "current": false,
+        "max": false,
+        "min": false,
+        "rightSide": false,
+        "show": true,
+        "total": false,
+        "values": false
+      },
+      "lines": true,
+      "linewidth": 1,
+      "nullPointMode": "null",
+      "options": {
+        "alertThreshold": true
+      },
+      "percentage": false,
+      "pluginVersion": "7.3.7",
+      "pointradius": 2,
+      "points": false,
+      "renderer": "flot",
+      "seriesOverrides": [],
+      "spaceLength": 10,
+      "stack": false,
+      "steppedLine": false,
+      "targets": [
+        {
+          "expr": "sum(apisix_shared_dict_free_space_bytes{name=~\"internal-status|upstream-healthcheck|worker-events|lrucache-lock|balancer-ewma|balancer-ewma-locks|balancer-ewma-last-touched-at|etcd-cluster-health-check\", instance=~\"$instance\"}) by (name)",
```

Comment:
We don't need to hardcode the name of each shdict, right?

BTW, we should provide an available percent (free / capacity) for each shdict than the sum. The sum is not helpful.

Reply:
> We don't need to hardcode the name of each shdict, right?
> 
> BTW, we should provide an available percent (free / capacity) for each shdict than the sum. The sum is not helpful.

I tryed to do it but failed, would you give me some suggestion(like use which function in grafana) in caculate the percent (free/capacity) for each shdict, thank you very much ! : )

Reply:
You can learn about the syntax of PromQL and try for it. AFAIK, a division between free and capacity is enough.